File: src/main/java/com/hf/hfw/api/v1/AccountService.java
package com.hf.hfw.api.v1;

import com.hf.homefinanceshared.Account;
import java.util.List;
import javax.ws.rs.Consumes;
import javax.ws.rs.DELETE;
import javax.ws.rs.GET;
import javax.ws.rs.POST;
import javax.ws.rs.Path;
import javax.ws.rs.PathParam;
import javax.ws.rs.Produces;

/**
 *
 * @author phillip.dorrell
 */
public interface AccountService {
    @GET
    @Produces({"application/json"})
    @Consumes({"application/json"})
    @Path("/search/all")    
    public List<Account> getAccounts();
    
    @GET
    @Produces({"application/json"})
    @Consumes({"application/json"})
    @Path("/getbyId/{accountId}")    
    public Account getById(@PathParam("accountId") String accountId);
    
    @GET
    @Produces({"application/json"})
    @Consumes({"application/json"})
    @Path("/getbyName/{name}")    
    public Account getByName(@PathParam("name") String name);
    
    
    @GET
    @Produces({"application/json"})
    @Consumes({"application/json"})
    @Path("/getbyExternalId/{externalId}")    
    public Account getByExternalId(@PathParam("externalId") String externalId);
    
    
    @POST
    @Produces({"application/json"})
    @Consumes({"application/json"})
    @Path("/save")    
    public Account saveAccount(Account account) throws Exception;
    
    @DELETE
    @Produces({"application/json"})
    @Consumes({"application/json"})
    @Path("/delete/{accountId}")
    public void deleteAccount(@PathParam("accountId") String accountId); 
    
      
}

File: src/main/webapp/script/app/services/accountservice.js
angular.module('AccountServiceModule',[]);
angular.module('AccountServiceModule').factory("AccountService", function ($http) {
    //var users = ["Peter", "Daniel", "Nina"]
    var AccountService = {};

    AccountService.getAccounts = function () {
        return $http({
            method: 'GET',
            url: '/services/api/v1/accounts/search/all'
        });
    };
    AccountService.getAccount = function (id) {
        return $http({
            method: 'GET',
            url: '/services/api/v1/accounts/getbyId/' + id
        });
    };
    AccountService.deleteAccount = function (id) {
        var csrfHeaderName = $("meta[name='_csrf_header']").attr("content");
        var csrfHeaderValue = $("meta[name='_csrf']").attr("content");

        return $http({
            method: 'DELETE',
            url: '/services/api/v1/accounts/delete/' + id,
            //data: JSON.stringify(account), // pass in data as strings
            headers: {'Content-Type': 'application/json', 'X-CSRF-TOKEN': csrfHeaderValue}  // set the headers so angular passing info as form data (not request payload)

        });
    };

    AccountService.saveAccount = function (account) {
        delete account.$$hashKey;
        var csrfHeaderName = $("meta[name='_csrf_header']").attr("content");
        var csrfHeaderValue = $("meta[name='_csrf']").attr("content");
        return $http({
            method: 'POST',
            url: '/services/api/v1/accounts/save',
            data: JSON.stringify(account), // pass in data as strings
            headers: {'Content-Type': 'application/json', 'X-CSRF-TOKEN': csrfHeaderValue}  // set the headers so angular passing info as form data (not request payload)
        })
    };

    return AccountService;
});
